WordPress/wporg-parent-2021
# Block styles: Update search block to match new theme designs

See https://github.com/WordPress/wporg-mu-plugins/issues/419 — the search field's styling is being updated in the next iteration of the redesign. This PR updates the current search styles to use this new design.

- Remove the "secondary style" since the old search styles are deprecated now (correct?)
- Adds a new style for "on dark" to handle the dark form
- The focus states are a little different than the mockup, since the icon-button and input are separate, both focusable elements. There's a video at the end of this to show what I mean. Happy to iterate on this.

Mockup:

<img width="581" alt="Screenshot 2023-08-31 at 6 16 20 PM" src="https://github.com/WordPress/wporg-parent-2021/assets/541093/63ffe558-7848-4c0c-bc63-625f98d30404">

### Screenshots

This will impact live sites using the parent theme, I think it's minimal (and expected, if this is the new search design that should be used). If that's not the case, we should figure out what cases use which search style. 

| Before | After |
|--------|-------|
| ![prod-showcase](https://github.com/WordPress/wporg-parent-2021/assets/541093/05763a98-1d28-4a94-9245-5955735d33d9) | ![search-showcase](https://github.com/WordPress/wporg-parent-2021/assets/541093/b0a805e4-fce2-4e91-8387-34648d8e2813) |
| ![prod-dev-blog](https://github.com/WordPress/wporg-parent-2021/assets/541093/88342157-2225-43d5-b0e4-119e1981b5f8) | ![search-dev-blog](https://github.com/WordPress/wporg-parent-2021/assets/541093/e3dfd508-8f7f-4a16-b0c1-9e36ea0e5fe4) |
| ![prod-docs](https://github.com/WordPress/wporg-parent-2021/assets/541093/f34a28d4-baf6-4e8e-adfc-c120d4f5c984) | ![search-docs](https://github.com/WordPress/wporg-parent-2021/assets/541093/904ff809-842d-4b8f-9600-7b79ee593bb4) |
| ![prod-docs-single](https://github.com/WordPress/wporg-parent-2021/assets/541093/0b9abe5f-df35-4db5-a9fd-fb0366c0cd9d) | ![search-docs-single](https://github.com/WordPress/wporg-parent-2021/assets/541093/189d76cf-b52f-4b66-b06e-8cd1b9ade819) |

The following video shows the various focus, hover, and filled-in states. I know most of these variations won't be used, but on the chance they are, it shouldn't break.

https://github.com/WordPress/wporg-parent-2021/assets/541093/462c97be-0f8b-4b30-bd41-16f75c9eb3b0

### How to test the changes in this Pull Request:

1. Use this branch for the parent theme, load any child theme project (showcase, developer, documentation, etc)
2. View any existing search blocks
3. Add search to a page, both styles & various configurations should work
